test/unit/math/rev/arr/functor/coupled_mm.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
```

Comment:
Let's use our standard header guard here.
`TEST_UNIT_MATH_REV_ARR_FUNCTOR_COUPLED_MM_HPP`

Questions about naming:
- what about this is "coupled"?
- could we just call it michaelis menten instead of mm?